extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+from typing import Any, Dict, Literal, Optional
+
+import firebase_admin
+from firebase_admin import credentials, firestore_async
+from google.cloud import firestore
+from google.cloud.firestore_v1.base_query import FieldFilter
+from pydantic import BaseModel
+
+import models
+
+from .. import datastore
+
+
+class Config(BaseModel, datastore.AbstractConfig):
+    kind: Literal["firestore"]
+    projectId: str
+    serviceAccountId: str
+
+
+class Client(datastore.Client[Config]):
+    __client: firestore.AsyncClient
+
+    @datastore.classproperty
+    def kind(cls):
+        return "firestore"
+
+    def __init__(self, client: firestore.AsyncClient):
+        self.__client = client
+
+    @classmethod
+    async def create(cls, config: Config) -> "Client":
+        cred = credentials.ApplicationDefault()
+        firebase_admin.initialize_app(
+            cred,
+            options={
+                "projectId": config.projectId,
+                "serviceAccountId": config.serviceAccountId,
+            },
+        )
```

Comment:
Are you sure we need to do this? Is this creating a new app or just representing the App locally?

Reply:
Deleted